sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java

```diff
@@ -420,6 +426,37 @@ public static DataflowPipelineRunner fromOptions(PipelineOptions options) {
     return super.apply(new AssignWindows<>(transform), input);
   }
 
+  @Nullable
+  private void maybeRegisterDebuggee(DataflowPipelineOptions options, String uniquifier) {
+    if (!options.getEnableCloudDebugger() || options.getDebuggee() != null) {
+      return;
+    }
+
+    Clouddebugger debuggerClient = Transport.newClouddebuggerClient(options).build();
+    RegisterDebuggeeRequest registerReq = new RegisterDebuggeeRequest();
+    registerReq.setDebuggee(new Debuggee()
+        .setProject(options.getProject())
+        .setUniquifier(uniquifier)
+        .setDescription(uniquifier)
+        .setAgentVersion("google.com/cloud-dataflow-java/v1"));
+
+    try {
+      RegisterDebuggeeResponse registerResponse =
+          debuggerClient.controller().debuggees().register(registerReq).execute();
+      Debuggee debuggee = registerResponse.getDebuggee();
+      if (debuggee.getStatus() != null && debuggee.getStatus().getIsError()) {
+        LOG.error("Error registering with debugger: {}",
+            debuggee.getStatus().getDescription().getFormat());
+        return;
+      }
+
+      options.setDebuggee(debuggee);
+    } catch (IOException e) {
+      System.out.println("Unable to register with debugger: " + e);
```

Comment:
I'm torn. The current behavior of the worker will be to register a debuggee using the Job ID if the flag to enable the debugger is present but this call fails -- so in that regard there is a graceful fallback. On the other hand, I think failing here wouldn't be unreasonable since debugging was requested, and this error may indicate a problem that the worker won't be able to deal with.

I'll add this tomorrow.